### DISCUSSION TITLE
Adding different models

### DISCUSSION THREAD

**Discussion by adinath1233** [deepdoctection/deepdoctection#352] [Q&A] [answered]
Hey so I have taken the pull of the repository I want to use different model in for layouting or drawing bounding box that is from LayoutParser TableBank for table purposes so how I can add that in deepdoctection. And I want to set in config file like USE_TABLE_DETECTOR=True something like this.

**Comment by JaMe76** [comment]
There is a tutorial for this [here](https://deepdoctection.readthedocs.io/en/latest/tutorials/running_pre_trained_models_from_third_party_libraries_notebook/).

If there are more questions I am happy to answer them.

**Reply by adinath1233** [reply]
> There is a tutorial for this [here](https://deepdoctection.readthedocs.io/en/latest/tutorials/running_pre_trained_models_from_third_party_libraries_notebook/).
> 
> If there are more questions I am happy to answer them.

Hi @JaMe76,
Thanks for replying appreciated. The only doubt I have is that if I am making changes in a config file and registering in a pulled or cloned repository will the analyzer work with that?

**Reply by JaMe76** [reply, marked as answer]
Sorry for late reply.

If you register the model first and then instantiate the analyzer with config changes then it will work. 

E.G.

```python

import deepdoctection as dd



if __name__ == "__main__":

    dd.ModelCatalog.register("layoutparser/publaynet/model_final.pth", dd.ModelProfile(
        name="layoutparser/publaynet/model_final.pth",
        description="Add some infos regarding the model",
        config="layoutparser/publaynet/config.yml",
        tp_model=False,
        size=[],
        categories={
            1: dd.LayoutType.TEXT,
            2: dd.LayoutType.TITLE,
            3: dd.LayoutType.LIST,
            4: dd.LayoutType.TABLE,
            5: dd.LayoutType.FIGURE,
        },
        model_wrapper="D2FrcnnDetector",
    ))

    # you must store layoutparser config file and weights as returned from the following statement
    print(dd.ModelCatalog.get_full_path_weights("layoutparser/publaynet/model_final.pth"))  
    print(dd.ModelCatalog.get_full_path_configs("layoutparser/publaynet/model_final.pth"))


    path = "/path/to/your/pdf/file.pdf"

    # replace the default layout model with the new registered layoutparser model
    analyzer = dd.get_dd_analyzer(config_overwrite=["PT.LAYOUT.WEIGHTS=layoutparser/publaynet/model_final.pth",])
    df = analyzer.analyze(path=path)

    df.reset_state()

    for dp in df:
        dp.viz(interactive=True)
```

**Reply by adinath1233** [reply]
Hey, thank you for the reply I got it how it works appreciated it.
